kitsune/users/auth.py

```diff
@@ -149,3 +151,64 @@ def update_user(self, user, claims):
 
         profile.save()
         return user
+
+    def authenticate(self, request, **kwargs):
+        """Authenticate a user based on the OIDC/oauth2 code flow.
+
+        Users in Kitsune can start the oauth/oidc flow when they are already authenticated.
+        In this case we have a user who is trying to connect the current SuMo account with FxA.
+        If the user is not authenticated we have a login/signup attempt.
+        """
+
+        # If the request has the /oidc/callback/ path then probably there is a login
+        # attempt in the admin interface. In this case just return None and let
+        # the OIDC backend handle this request.
+        if request.path == u'/oidc/callback/':
```

Comment:
@pmac @johngian Thoughts on this? Any ideas on how to do this differently? Basically what we want here is to skip the FxA auth backend if the login is targeting the admin interface.